Diplom_3

ssvasilev opened the pull request.

ssvasilev committed:
Настроил pom.xml, завершил UI тесты регистрации

ssvasilev committed:
Настроил pom.xml, завершил UI тесты регистрации

ssvasilev committed:
Добавил классы PageObject ещё для нескольких страниц, покрыл тестами авторизацию и переходы в личный кабинет и из личного кабинета

ssvasilev committed:
Добавил тесты на переходы к разделам Булки, Соусы, Начинки

ssvasilev committed:
Выполнил финальный прогон, добавил отчёт Allure

ssvasilev committed:
Delete .idea directory

ssvasilev committed:
Переименовал класс PageObject

ssvasilev committed:
Merge remote-tracking branch 'origin/develop3' into develop3

pavel-ulan commented on src/test/java/LoginTest.java:
```diff
@@ -0,0 +1,110 @@
+        UserAPI.createUser("autotestvasilevss@yandex.ru", "Сергей", "q1w2e3r4t5");
+        // создаём драйвер для браузера Chrome
+        ChromeOptions options = new ChromeOptions();
+        //options.addArguments("--no-sandbox", "--headless", "--disable-dev-shm-usage");
```
**можно лучше**
не оставляй неинформативных комментов.

pavel-ulan commented on src/test/java/LoginTest.java:
```diff
@@ -0,0 +1,110 @@
+        //options.addArguments("--no-sandbox", "--headless", "--disable-dev-shm-usage");
+        driver = new ChromeDriver(options);
+        // переходим на страницу тестового приложения
+        driver.get("https://stellarburgers.nomoreparties.site/");
```
**можно лучше**
насколько правильно делать это в before методе (это и инициализации драйвера касается и открытия страницы). Да, так тоже можно, конечно, но подумай, возможно найдешь более "красивое" решение

pavel-ulan commented on src/test/java/LoginTest.java:
```diff
@@ -0,0 +1,110 @@
+        WebElement dynamicElement = (new WebDriverWait(driver, 10))
+                .until(ExpectedConditions.presenceOfElementLocated(By.xpath(".//button[contains(text(),'Оформить заказ')]")));
+        //Сравниваем текущий URL с адресом главной страницы
+        Assert.assertEquals("https://stellarburgers.nomoreparties.site/",driver.getCurrentUrl());
```
**надо исправить**
и что ты этим проверишь? у главной страницы такой-же URI. Даже если тест пройдет, ты не можешь на 100% гарантировать, что логин был выполнен

pavel-ulan replied:
проверка через "оформить заказ" вполне валидная, только ее надо оформить как ожидание, а то сейчас в случае неудачи будет просто exception.
И еще, если хочешь использовать несколько ассертов в тесте, пользуйся softAsserts

pavel-ulan commented on src/test/java/LoginTest.java:
```diff
@@ -0,0 +1,110 @@
+        // Закрой браузер
+        driver.quit();
+        //Удали тестового пользователя
+        UserAPI.deleteUser("autotestvasilevss@yandex.ru","q1w2e3r4t5");
```
**молодец**
да, использовать удаление через API отличное решение - молодец

pavel-ulan commented on src/test/java/SectionsTest.java:
```diff
@@ -0,0 +1,64 @@
+        mainPage.sectionBunButtonClick(); //Нажимаем на кнопку "Булки"
+        //Ждём, пока появится раздел Булки
+        WebElement dynamicElement = (new WebDriverWait(driver, 10))
+                .until(ExpectedConditions.presenceOfElementLocated(By.xpath(".//h2[contains(text(),'Булки')]")));
```
**надо исправить**
все хорошо, только твой локатор всегда present даже если ты в другом разделе находишься =)
так что он у тебя будет всегда true.
попереходи по разделам и посмотри как меняются локаторы и построй тест на основе проверки "в каком разделе я нахожусь"

pavel-ulan commented on src/test/java/page_object/LoginPage.java:
```diff
@@ -0,0 +1,65 @@
+    private By forgotPasswordButton = By.xpath(".//a[contains(text(),'Восстановить пароль')]");
+
+
+    //public static By loingTitle = By.xpath(".//div[@class='Auth_login__3hAey']/h2");
```
?

pavel-ulan reviewed (changes requested):
Неплохо, есть пара ошибок в том числе в тестах. посмотри комменты и поправь

ssvasilev committed:
Убрал лишние комменты, вынес инициализацию браузера в общий класс, от которого наследуются остальные.
Изменил проверку на кнопку "Оформить заказ",
Перестроил тесты секций на изменяющиеся локаторы.

Почистил код от лишних before и импортов

pavel-ulan commented on src/test/java/LoginTest.java:
```diff
@@ -0,0 +1,97 @@
+        //После успешного логина должно перенести на главную страницу и появиться кнопка "Оформить заказ"
+        WebElement dynamicElement = (new WebDriverWait(driver, 10))
+                .until(ExpectedConditions.presenceOfElementLocated(xpath(".//button[contains(text(),'Оформить заказ')]")));
+        Assert.assertTrue(driver.findElement(xpath(".//button[contains(text(),'Оформить заказ')]")).isEnabled()); //Убедись, что кнопка доступна
```
👍

pavel-ulan commented on src/test/java/RegistrationTest.java:
```diff
@@ -0,0 +1,64 @@
+        Assert.assertEquals("https://stellarburgers.nomoreparties.site/login",driver.getCurrentUrl());
+
+        //Удаляем тестового пользователя
+        UserAPI.deleteUser(UserAPI.testUserEmail,UserAPI.testUserPassword);
```
**можно лучше**
а после неуспешного? ситуация опять же: упал тест - не удалился пользователь

pavel-ulan commented on src/test/java/SectionsTest.java:
```diff
@@ -0,0 +1,60 @@
+
+    private String xpathBun = "//div[@class='tab_tab__1SPyG tab_tab_type_current__2BEPc pt-4 pr-10 pb-4 pl-10 noselect']/span[contains(text(),'Булки')]";
+    private String xpathSauce = "//div[@class='tab_tab__1SPyG tab_tab_type_current__2BEPc pt-4 pr-10 pb-4 pl-10 noselect']/span[contains(text(),'Соусы')]";
+    private String xpathIngredients = "//div[@class='tab_tab__1SPyG tab_tab_type_current__2BEPc pt-4 pr-10 pb-4 pl-10 noselect']/span[contains(text(),'Начинки')]";
```
**можно лучше**
не самое элегантное решение. первое - достаточно было просто проверять на содержит ли локатор `tab_tab_type_current__2BEPc` второе - держать локаторы в классе тесте - не самое хорошее решение

pavel-ulan reviewed (approved):
Ну... скажем так, мне не очень нравится решение, но технически ты исправил и все условно проходит
